pkg/providers/vsphere/vmprovider_vm_utils.go

```diff
@@ -740,3 +759,30 @@ func isVMPaused(vmCtx pkgctx.VirtualMachineContext) bool {
 	delete(vmCtx.VM.Labels, vmopv1.PausedVMLabelKey)
 	return false
 }
+
+func PatchSnapshotStatus(vmCtx pkgctx.VirtualMachineContext, k8sClient ctrlclient.Client,
+	objName string, objNs string, snapMoRef *vimtypes.ManagedObjectReference) error {
+	snapShot := &vmopv1.VirtualMachineSnapshot{}
+	objKey := ctrlclient.ObjectKey{Name: objName, Namespace: objNs}
+	// get snapshot again to ensure it's up-to-date.
+	err := k8sClient.Get(vmCtx, objKey, snapShot)
```

Comment:
Again, just curious, do we need this?  The Status.uniqueID, and status.Quiesced are not supposed to be patched/modified by any one else other than this controller.

Reply:
I think it is okay to remove this.  We can always add it if we start to see conflicts.  I don't expect multiple concurrent snapshot operations on VMs over and over anyway.